src/main/java/com/anaplan/client/ServerFile.java

```diff
@@ -325,6 +322,10 @@ public void upLoad(File source, boolean deleteExisting, int chunkSize) throws IO
                 System.arraycopy(buffer, 0, finalBuffer, 0, finalSize);
                 //calculating the total read size from the file
                 totalReadSoFar += finalSize;
+
+                if ((chunkIterator.hasNext() && finalBuffer.length == 0) || buffer.length == 0) {
```

Comment:
a comment explaining the purpose of this check might be helpful in future

Reply:
Sure will do that.